pool.go

```diff
@@ -24,28 +24,85 @@ import (
 )
 
 type modulePool[T interfaces.Signature] struct {
-	pool sync.Pool
-	new  func() (*module[T], error)
+	pool    sync.Pool
+	maxSize uint32
+	new     func(*module[T]) (*module[T], error)
```

Comment:
Why does `new` accept a `*module[T]` as a parameter? Seems like it should only return new modules.